src/components/mapPageComponents/mapView/index.tsx

```diff
@@ -54,225 +63,234 @@ const MapView: React.FC = () => {
+
+        // Creates a new layer
+        const blocksLayer = new google.maps.Data({ map });
+
+        // Loads the objects into the layer
+        blocksLayer.addGeoJson(blocks);
+
+        // Sets the style of the layer to colored blocks with black outline
+        function setBlocksStyle(v: boolean) {
+          blocksLayer.setStyle((feature) => {
+            let color = 'green';
+
+            // Use this for coloring reserved/completed blocks a different color
+            if (feature.getProperty('ID') % 10 === 0) {
+              color = 'yellow';
+            }
+
+            if (feature.getProperty('ID') % 10 === 1) {
+              color = 'red';
+            }
+
+            // Use this for selecting blocks
+            if (feature.getProperty('PRECINCT') === 'YES') {
+              color = 'black';
+            }
+
+            // Set styling here
+            return {
+              fillColor: color,
+              strokeColor: 'black',
+              strokeWeight: 1,
+              visible: v,
+            };
+          });
         }
 
-        // Use this for selecting blocks
-        if (feature.getProperty('PRECINCT') === 'YES') {
-          color = 'black';
+        // Initially false while the neighborhoods are shown
+        setBlocksStyle(false);
+
+        // Creates a new layer
+        const neighborhoodsLayer = new google.maps.Data({ map });
+
+        // Function: Returns name in shortHand
+        /*
+        function shortHand(name: string): string {
+          for (const shName of shortHandNames) {
+            if (shName.key === name) {
+              return shName.value;
+            }
+          }
+          return name;
+        }
+        */
+        // Loads the objects into the layer
+        neighborhoodsLayer.addGeoJson(neighborhoods);
+        /*
+          () => {
+            // For each feature in neighbourhoodsLayer, add a marker
+            // We need to do it here as the GeoJson needs to load first
+            neighborhoodsLayer.forEach((feature) => {
+              // If you want, check here for some constraints.
+              const marker = new google.maps.Marker({
+                map,
+                draggable: false,
+                label: {
+                  color: 'white',
+                  fontWeight: 'bold',
+                  text: shortHand(feature.getProperty('Name')),
+                },
+                // Removed the icon here, only text on map.
+                icon: {
+                  labelOrigin: new google.maps.Point(11, 50),
+                  url: '',
+                  size: new google.maps.Size(22, 40),
+                },
+                position: {
+                  lat: feature.getProperty('Latitude'),
+                  lng: feature.getProperty('Longitude'),
+                },
+              });
+              markersArray.push(marker);
+              marker.setMap(map);
+            });
+          },
+        );
+        */
```

Comment:
Do we want to keep this comment?

Reply:
We are going to need to reuse a lot of this logic, however because of performance concerns I'm pretty sure we do not just want to use it as is. @florisdobber do you want to keep this comment here?